Lib/test/test_ast.py

```diff
@@ -1227,21 +1227,20 @@ def test_dump(self):
         node = ast.parse('spam(eggs, "and cheese")')
         self.assertEqual(ast.dump(node),
             "Module(body=[Expr(value=Call(func=Name(id='spam', ctx=Load()), "
-            "args=[Name(id='eggs', ctx=Load()), Constant(value='and cheese')], "
-            "keywords=[]))], type_ignores=[])"
+            "args=[Name(id='eggs', ctx=Load()), Constant(value='and cheese')]))])"
         )
         self.assertEqual(ast.dump(node, annotate_fields=False),
             "Module([Expr(Call(Name('spam', Load()), [Name('eggs', Load()), "
-            "Constant('and cheese')], []))], [])"
+            "Constant('and cheese')]))])"
```

Comment:
We need to think more about the interaction between show_empty and annotate_fields=False.

The current behavior on this PR seems wrong:
```
>>> dump(arguments(args=[arg("x")]), annotate_fields=False)
"arguments([arg('x')])"
>>> dump(eval(_))
"arguments(posonlyargs=[arg(arg='x')])"
```

I think the right behavior should be that if annotate_fields=False, we show empty fields (even if show_empty=True) if there is a nonempty field following them.

Reply:
Done, fixed!